dipy/core/tests/test_gradients.py

```diff
@@ -1008,3 +1010,74 @@ def test_gradient_table_len():
         obt_grad_count1 = gtab.__len__()
         obt_grad_count2 = len(gtab)
         assert exp_grad_count == obt_grad_count1 == obt_grad_count2
+
+
+def test_extract_b0():
+    dwi = np.random.rand(10, 10, 10, 5).astype(np.float32)
+    b0_mask = np.array([True, False, True, True, False])
+    expected_first = dwi[..., 0]
+    expected_mean = np.mean(dwi[..., [0, 2, 3]], axis=-1)
+    expected_all = dwi[..., [0, 2, 3]]
+
+    # Test extracting all b0 volumes
+    result_all = extract_b0(dwi, b0_mask, strategy="all")
+    npt.assert_equal(result_all.shape[-1], np.sum(b0_mask))
+    npt.assert_array_equal(result_all, expected_all)
+
+    # Test extracting first b0 volume
+    result_first = extract_b0(dwi, b0_mask, strategy="first")
+    npt.assert_equal(result_first.shape, dwi.shape[:-1])
+    npt.assert_array_equal(result_first, expected_first)
+
+    # Test extracting mean b0 volume
+    result_mean = extract_b0(dwi, b0_mask, strategy="mean")
+    npt.assert_equal(result_mean.shape, dwi.shape[:-1])
+    npt.assert_almost_equal(result_mean, expected_mean)
+
+    # Test contiguous grouping
+    result_grouped = extract_b0(dwi, b0_mask, group_contiguous_b0=True, strategy="mean")
+    assert len(result_grouped.shape) >= 3  # Should be at least 3D
```

Comment:
Why are we only testing shape here instead of the expected values like the one above?